src/iidm/Network.cpp

```diff
@@ -83,7 +84,27 @@ Network::Network(Network&& network) noexcept :
     m_networkIndex(*this, std::move(network.m_networkIndex)),
     m_variantManager(*this, std::move(network.m_variantManager)),
     m_busBreakerView(*this),
-    m_busView(*this) {
+    m_busView(*this),
+    m_variants(std::move(network.m_variants)) {
+    m_variants.setVariantManagerHolder(*this);
```

Comment:
I think we can try to use the same design than for NetworkIndex and VariantManager